aces/idt/core/structures.py

```diff
@@ -0,0 +1,197 @@
+"""Module contains a selection of helper classes used throughout the IDT
+
+"""
+import json
+from pathlib import Path
+
+import numpy as np
+
+
+class PathEncoder(json.JSONEncoder):
+    """
+    A custom JSONEncoder subclass that knows how to encode Path objects.
+    """
+
+    def default(self, obj):
+        """
+        Convert the object to a JSON serializable object
+
+        Parameters
+        ----------
+        obj : object we want to serialize
+
+        Returns the JSON serializable object
+        -------
+
+        """
+        if isinstance(obj, Path):
+            # Convert the PosixPath to a string
+            return str(obj)
+        if isinstance(obj, np.ndarray):
+            return list(obj)
+        return super().default(obj)
+
+
+class SerializableConstants:
+    """Constants for the serializable"""
+
+    HEADER = "header"
+    DATA = "data"
+
+
+class IDTMetaData:
```

Comment:
**Major**

Seems like this wants to be a `dataclass`.

Reply:
Are you referring to the IDTMetaData? or the SerializableConstants? or both?

Not used dataclass much before whats the benefit?

The SerializableConstants is essentially an Enum which probably should use the Enum python type

Reply:
Resolved By Thomas